Casperf fixes

tiagolobocastro opened the pull request:
build(container/images): reduce container image size
    
    Fixup the spdk rpath in the pre-fixup allowing for the default
    fixup phase to shrink.
    Split io-engine derivation into 2 outputs, to avoid pulling in all binaries.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    feat: add casperf container image
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    fix(io-engine-client): use MY_POD_IP is present
    
    When running from k8s pod, we should not have to specify our own grpc endpoint.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    chore: update auto assign list
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    fix(casperf/lvs): purge lvs by wiping the first 8MiB
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    fix(null_bdev): parse size with 64bit
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    chore: allow specifying release version of libspdk
    
    Allows us to test performance within a dev shell.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
chore: allow specifying release version of libspdk

Allows us to test performance within a dev shell.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
fix(null_bdev): parse size with 64bit

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
fix(casperf/lvs): purge lvs by wiping the first 8MiB

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
chore: update auto assign list

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

dsharma-dc commented on io-engine/src/bdev/lvs.rs:
Could you share an example how the LvsMode is passed into by casperf tool?
 Is the Purge mode only with casperf?

tiagolobocastro replied:
`sudo casperf "lvol:///lvol-1?size=220GiB&lvs=lvs:///pool-1?mode=purge&disk=aio:///dev/sda"`
Yes

tiagolobocastro committed:
fix(io-engine-client): use MY_POD_IP is present

When running from k8s pod, we should not have to specify our own grpc endpoint.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
feat: add casperf container image

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
build(container/images): reduce container image size

Fixup the spdk rpath in the pre-fixup allowing for the default
fixup phase to shrink.
Split io-engine derivation into 2 outputs, to avoid pulling in all binaries.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
bors cancel

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
bors merge

dsharma-dc commented on n.sh:
Is this script `n.sh` merged by mistake? :)

tiagolobocastro replied:
doh, must have git added by mistake

dsharma-dc replied:
you'll remove it in a separate PR I'd hope.

abhilashshetty04 commented on io-engine/src/bdev/lvs.rs:
```diff
@@ -233,42 +233,60 @@ impl Lvs {
             name: self.name.to_owned(),
         })
     }
 
```
@tiagolobocastro documentation?
